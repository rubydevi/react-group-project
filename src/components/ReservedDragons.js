import React from 'react';

const ReservedDragons = () => (
  <div>ReservedDragons</div>
);

export default ReservedDragons;
