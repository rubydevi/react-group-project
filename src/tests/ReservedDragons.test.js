import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import ReservedDragons from '../components/ReservedDragons';

const mockStore = configureStore([]);

test('ReservedDragons component renders correctly with reserved dragons', () => {
  const store = mockStore({
    dragons: {
      dragons: [
        {
          id: '1',
          name: 'Dragon 1',
          reserved: true,
        },
        {
          id: '2',
          name: 'Dragon 2',
          reserved: true,
        },
      ],
    },
  });

  const tree = renderer.create(
    <Provider store={store}>
      <ReservedDragons />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('ReservedDragons component renders correctly with no reserved dragons', () => {
  const store = mockStore({
    dragons: {
      dragons: [
        {
          id: '3',
          name: 'Dragon 3',
          reserved: false,
        },
      ],
    },
  });

  const tree = renderer.create(
    <Provider store={store}>
      <ReservedDragons />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
