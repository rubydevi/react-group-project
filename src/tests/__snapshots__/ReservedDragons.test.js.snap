// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservedDragons component renders correctly with no reserved dragons 1`] = `
<div
  className="mt-4 container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h5>
        Dragons
         
        <span
          className="badge bg-info"
        >
          No dragon reserved
        </span>
      </h5>
    </div>
  </div>
</div>
`;

exports[`ReservedDragons component renders correctly with reserved dragons 1`] = `
<div
  className="mt-4 container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          Dragon 1
        </div>
        <div
          className="list-group-item"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          Dragon 2
        </div>
      </div>
    </div>
  </div>
</div>
`;
